### PR TITLE
[BEAM-1628] Allow empty port for flink master url

### PR DESCRIPTION
**Pull request by mxm**
This allows only specifying the host name without a port. It reads the port from
the configuration or uses the default Flink port if non configured. It does not
fall back to the special "[auto]" mode anymore if the port is omitted.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by angoenka** [comment]
Run Portable_Python PreCommit

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java [hunk 1]
```diff
@@ -77,17 +76,23 @@ static ExecutionEnvironment createBatchExecutionEnvironment(
...
     } else if ("[auto]".equals(masterUrl)) {
       flinkBatchEnv = ExecutionEnvironment.getExecutionEnvironment();
-    } else if (masterUrl.matches(".*:\\d*")) {
-      List<String> parts = Splitter.on(':').splitToList(masterUrl);
+    } else {
+      String[] hostAndPort = masterUrl.split(":", 2);
+      final String host = hostAndPort[0];
+      final int port;
+      if (hostAndPort.length > 1) {
+        port = Integer.parseInt(hostAndPort[1]);
```
Comment (angoenka) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java]: Shall we catch the integer parsing exception and log 
`     LOG.warn("Unrecognized Flink Master URL {}", masterUrl);	`

Comment (mxm) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkExecutionEnvironments.java]: If the port can't be parsed, then something is wrong. I would just throw an exception with a meaningful error message. 

As a side note, I realized we don't support IPv6 Ip addresses.

**Comment by mxm** [comment]
Run Portable_Python PreCommit

**Comment by mxm** [comment]
Run Flink ValidatesRunner

**Pull request merged**